src/lib/addons/feature-event-formatter-md.ts

```diff
@@ -82,307 +21,39 @@ export enum LinkStyle {
-        action: '*{{user}}* created banner *{{event.data.message}}*',
-        path: '/admin/message-banners',
-    },
-    [BANNER_DELETED]: {
-        label: 'Banner deleted',
-        action: '*{{user}}* deleted banner *{{event.preData.message}}*',
-        path: '/admin/message-banners',
-    },
-    [BANNER_UPDATED]: {
-        label: 'Banner updated',
-        action: '*{{user}}* updated banner *{{event.preData.message}}*',
-        path: '/admin/message-banners',
-    },
-    [PROJECT_CREATED]: {
-        label: 'Project created',
-        action: '*{{user}}* created project *{{project}}*',
-        path: '/projects',
-    },
-    [PROJECT_DELETED]: {
-        label: 'Project deleted',
-        action: '*{{user}}* deleted project *{{event.project}}*',
-        path: '/projects',
-    },
-    [SEGMENT_CREATED]: {
-        label: 'Segment created',
-        action: '*{{user}}* created segment *{{event.data.name}}*',
-        path: '/segments',
-    },
-    [SEGMENT_DELETED]: {
-        label: 'Segment deleted',
-        action: '*{{user}}* deleted segment *{{event.preData.name}}*',
-        path: '/segments',
-    },
-    [SEGMENT_UPDATED]: {
-        label: 'Segment updated',
-        action: '*{{user}}* updated segment *{{event.preData.name}}*',
-        path: '/segments',
-    },
-    [SERVICE_ACCOUNT_CREATED]: {
-        label: 'Service account created',
-        action: '*{{user}}* created service account *{{event.data.name}}*',
-        path: '/admin/service-accounts',
-    },
-    [SERVICE_ACCOUNT_DELETED]: {
-        label: 'Service account deleted',
-        action: '*{{user}}* deleted service account *{{event.preData.name}}*',
-        path: '/admin/service-accounts',
-    },
-    [SERVICE_ACCOUNT_UPDATED]: {
-        label: 'Service account updated',
-        action: '*{{user}}* updated service account *{{event.preData.name}}*',
-        path: '/admin/service-accounts',
-    },
-    [USER_CREATED]: {
-        label: 'User created',
-        action: '*{{user}}* created user *{{event.data.name}}*',
-        path: '/admin/users',
-    },
-    [USER_DELETED]: {
-        label: 'User deleted',
-        action: '*{{user}}* deleted user *{{event.preData.name}}*',
-        path: '/admin/users',
-    },
-    [USER_UPDATED]: {
-        label: 'User updated',
-        action: '*{{user}}* updated user *{{event.preData.name}}*',
-        path: '/admin/users',
-    },
-};
+type FormatStyle = 'simple' | 'markdown';
+
+interface IFeatureEventFormatterMdArgs {
+    unleashUrl: string;
+    linkStyle?: LinkStyle;
+    formatStyle?: FormatStyle;
+}
 
 export class FeatureEventFormatterMd implements FeatureEventFormatter {
     private readonly unleashUrl: string;
 
     private readonly linkStyle: LinkStyle;
 
-    constructor(unleashUrl: string, linkStyle: LinkStyle = LinkStyle.MD) {
+    private readonly formatStyle: FormatStyle;
+
+    constructor({
+        unleashUrl,
+        linkStyle = LinkStyle.MD,
+        formatStyle = 'simple',
```

Comment:
Could we rename and just use the linkStyle parameter? Seems to be options `SLACK | MD` just as what formatStyle is for: bold with one asterisk slack-style or bold with two markdown-style

Reply:
I'm not sure we can. Other integrations, aside from Slack, expect proper markdown links, but that doesn’t necessarily mean they fully support proper markdown. E.g. [Microsoft Teams](https://support.microsoft.com/en-us/office/use-markdown-formatting-in-microsoft-teams-4d10bd65-55e2-4b2d-a1f3-2bebdcd2c772)

For now, I’d prefer to leave the links as they are and simply add this additional option.

Reply:
Ok fair enough, they serve the same purpose more or less and work more or less the same, but since they looked so different I wondered if perhaps they could be merged.

Reply:
I think it's a very good point and something I definitely want to come back to sometime in the future. For now, I think I just want to ignore the different and inconsistent markdown implementations across integrations and leave them alone, as long as I'm still able to get properly formatted markdown out of our markdown formatter 😄